### DISCUSSION TITLE
C++ PCH warning: header stop not at file scope.  An IntelliSense PCH file was not generated.

### DISCUSSION THREAD

**Discussion by vhnatiuk** [danielaparker/jsoncons#358] [Q&A]
I have added jsoncons as an submodule to my project. Trying to use it in my code, but when I added the include 
`#include <jsoncons_ext/bson/bson.hpp>` to my `.h` or `.cpp` file, I'm getting error from the discussion title in the code editor, and a lot of compile errors after build of the project from the` include\jsoncons\detail\optional.hpp` file.

I have added the path to the  jsoncons `include` directory to the project Additional include Directories.
I'm using VS 2019. 
![image](https://user-images.githubusercontent.com/23076263/165333466-87d6b087-12a6-473c-96db-909d32e4d6d8.png)

**Comment by danielaparker** [comment]
Can you confirm which C++ Language Standard you're using? Is it C++14? Are you using C++14 consistently?

The fact that these errors are seen in `detail/optional.hpp`, and that you're using precompiled headers, makes me think that there's an issue with consistency in language standards selected. Inclusion of `detail/optional.hpp` is #defined out altogether if C++17 is selected.

**Reply by vhnatiuk** [reply]
> Can you confirm which C++ Language Standard you're using? Is it C++14? Are you using C++14 consistently?
> 
> The fact that these errors are seen in `detail/optional.hpp`, and that you're using precompiled headers, makes me think that there's an issue with consistency in language standards selected. Inclusion of `detail/optional.hpp` is #defined out altogether if C++17 is selected.

Yes, I'm using C++ 14

**Reply by vhnatiuk** [reply]
Also I tried the C++ 17 and C++ 20 and I have the same issue.

**Reply by danielaparker** [reply]
You couldn't have the _same_ issue if you compiled with C++17 or C++20, for those settings, the `detail/optional.hpp` would never be included, you could delete that file and it wouldn't make any difference for compilation, I just did, and it didn't. If you're seeing an error related to `detail/optional.hpp` , than you're not consistently compiling with C++17 or C++20.

**Reply by vhnatiuk** [reply]
![image](https://user-images.githubusercontent.com/23076263/165385310-2724b70f-b6c7-4c97-b401-04260307a06d.png)

![image](https://user-images.githubusercontent.com/23076263/165384991-35a255cf-3e83-4dac-b728-a9f42bfc8a35.png)

**Reply by vhnatiuk** [reply]
Yes, two more errors appear related with byte. I'm trying find out how to resolve it.

**Reply by vhnatiuk** [reply]
When I remove the include of any of jsoncons header, my project build with no error.
